requests_kerberos/kerberos_.py

```diff
@@ -11,10 +11,29 @@
 from requests.structures import CaseInsensitiveDict
 from requests.cookies import cookiejar_from_dict
 
-from .exceptions import MutualAuthenticationError
+from .exceptions import MutualAuthenticationError, KerberosExchangeError
 
 log = logging.getLogger(__name__)
 
+# a log sniffing context manager for catching logged errors on a specific call
+class SniffLog(logging.Handler):
+    def __init__(self, log_to_sniff, sniff_min_level=logging.ERROR):
+        logging.Handler.__init__(self)
+        self.log_to_sniff = log_to_sniff
+        self.stored_logs = []
+
+        self.setLevel(sniff_min_level)
+
+    def __enter__(self, **kwargs):
+        self.log_to_sniff.addHandler(self)
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.log_to_sniff.removeHandler(self)
+
+    def emit(self, record):
+        self.stored_logs.append(record.getMessage())
+
```

Comment:
This is a really clever piece of code, but I'm disinclined to use it.

My particular worries are around having multiple co-operating threads (green or otherwise) that both run through this code, leading to really unclear exception output. Is there really no way to get a better exception output than this?